### PR TITLE
Add smoke testing for MacOS

### PR DESCRIPTION
**Pull request by DanilBaibak**
Add smoke testing for MacOS:

- [x] Mac wheels
- [x] M1 wheels

### CODE REVIEW

**Comment by huydhn** [comment]
Paste a note here on how `mxschmitt/action-tmate@v3` overwrites the `authorized_keys` file on M1 runner https://github.com/pytorch/test-infra/issues/841 :)

File: .github/workflows/test_build_conda_m1.yml [hunk 1]
```diff
@@ -40,7 +41,7 @@ jobs:
     with:
       conda-package-directory: ${{ matrix.conda-package-directory }}
       repository: ${{ matrix.repository }}
-      ref: nightly
+      ref: ${{ matrix.ref }}
```
Comment (osalpekar) [inline .github/workflows/test_build_conda_m1.yml:44]: let's revert this back to nightly. (and all such changes in the other test workflow)

**Review by osalpekar — COMMENTED** [review]
LGTM. will approve after code cleanup. Once we merge this in, we'll have parity with circleCI. After syncing with audio team, we'll have full confirmation on whether we have all the needed functionality.

File: .github/workflows/build_wheels_macos.yml [hunk 2]
```diff
@@ -130,6 +144,40 @@ jobs:
...
+            echo $FFMPEG_ROOT
+            export DYLD_FALLBACK_LIBRARY_PATH="${PWD}/${{ inputs.repository }}/third_party/ffmpeg"
+            ls -la $DYLD_FALLBACK_LIBRARY_PATH
+            ${CONDA_RUN} DYLD_FALLBACK_LIBRARY_PATH="${PWD}/${{ inputs.repository }}/third_party/ffmpeg"
+
+            ${CONDA_RUN} python3 "${{ inputs.repository }}/${SMOKE_TEST_SCRIPT}"
+          fi
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+        if: ${{ failure() && runner.arch == "x86_64" }}
```
Comment (seemethere) [inline .github/workflows/build_wheels_macos.yml:176]:
```suggestion
        if: ${{ failure() && runner.arch == 'x86_64' }}
```

**Review by osalpekar — APPROVED** [review]
This looks great! Thanks for powering through some of these tricky build issues :)

**Pull request merged**